core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -0,0 +1,230 @@
+package cats
+package data
+
+import scala.annotation.tailrec
+import scala.collection.immutable.VectorBuilder
+import scala.util.Try
+import cats.instances.vector._
+
+/**
+ * A data type which represents a non empty Vector.
+ */
+final case class NonEmptyVector[A] private (toVector: Vector[A]) {
+
+  /** Gets the element at the index, if it exists */
+  def get(i: Int): Option[A] =
+    Try(getUnsafe(i)).toOption
```

Comment:
I think that `toVector.lift(i)` would probably be a more efficient implementation (though this is performance speculation without a benchmark).